python/src/nanoarrow/_ipc_lib.pyx

```diff
@@ -50,22 +50,32 @@ cdef extern from "nanoarrow_ipc.h" nogil:
 
 
 cdef class PyInputStreamPrivate:
-    cdef object obj
-    cdef object obj_method
-    cdef void* addr
-    cdef Py_ssize_t size_bytes
-    cdef int close_stream
+    cdef object _obj
+    cdef bint _close_stream
+    cdef void* _addr
+    cdef Py_ssize_t _size_bytes
 
     def __cinit__(self, obj, close_stream=False):
-        self.obj = obj
-        self.obj_method = obj.readinto
-        self.addr = NULL
-        self.size_bytes = 0
-        self.close_stream = close_stream
+        self._obj = obj
+        self._close_stream = close_stream
+        self._addr = NULL
+        self._size_bytes = 0
+
+    @property
+    def obj(self):
+        return self._obj
+
+    @property
+    def close_stream(self):
```

Comment:
Why does this object hold `close_stream` state? Do we not want `py_input_stream_release` to always close the stream?